ui-frontend/packages/catalog-ui-search/src/main/webapp/component/query-time/query-time.view.tsx

```diff
@@ -0,0 +1,201 @@
+import FilterInput from '../../react-component/filter/filter-input'
+export interface BasicFilterClass extends Omit<FilterClass, 'property'> {
+  property: string[]
+}
+
+type QueryTimeProps = {
+  value: undefined | BasicFilterClass
+  onChange: (e: any) => void
+}
+
+const getPossibleProperties = () => {
+  return metacardDefinitions.sortedMetacardTypes
+    .filter(
+      (definition: any) => !definition.hidden && definition.type === 'DATE'
+    )
+    .map((definition: any) => ({
+      label: definition.alias || definition.id,
+      value: definition.id,
+    })) as [
+    {
+      label: string
+      value: string
+    }
+  ]
+}
+
+const determinePropertiesToApplyTo = ({
+  value,
+}: {
+  value: BasicFilterClass
+}): Array<{ label: string; value: string }> => {
+  if (value.property) {
+    return value.property.map(property => {
+      return {
+        label: TypedMetacardDefs.getAlias({ attr: property }),
+        value: property,
+      }
+    })
+  } else {
+    return (properties.basicSearchTemporalSelectionDefault || []).map(
+      (property: string) => {
+        return {
+          label: TypedMetacardDefs.getAlias({ attr: property }),
+          value: property,
+        }
+      }
+    )
+  }
+}
+
+const QueryTime = ({ value, onChange }: QueryTimeProps) => {
+  React.useEffect(
+    () => {
+      if (value && value.property === undefined) {
+        onChange({
+          ...value,
+          property: determinePropertiesToApplyTo({ value }).map(
+            val => val.value
+          ),
+        })
+      }
+    },
+    [value]
+  )
+  if (value && value.property === undefined) {
+    return null // the use effect above should fire to take care of setting a default
+  }
+  return (
+    <>
+      <div>
+        <Typography className="pb-2">Time Range</Typography>
+
+        <TextField
+          fullWidth
+          variant="outlined"
+          size="small"
+          select
+          value={value ? value.type : 'any'}
+          onChange={e => {
+            if (e.target.value === 'any') {
+              onChange(undefined)
+            } else {
+              onChange({
+                ...value,
+                type: e.target.value,
+              })
+            }
+          }}
+        >
+          <MenuItem value="any">Any</MenuItem>
+          <MenuItem value="AFTER">After</MenuItem>
+          <MenuItem value="BEFORE">Before</MenuItem>
+          <MenuItem value="DURING">Between</MenuItem>
+          <MenuItem value="RELATIVE">Relative</MenuItem>
+        </TextField>
+      </div>
+      {value ? (
+        <div>
+          <Grid
+            container
+            alignItems="stretch"
+            direction="row"
+            wrap="nowrap"
+            className="pt-2"
+          >
+            <Grid item>
+              <Swath className="w-1 h-full" />
+            </Grid>
+            <Grid item className="w-full pl-2">
+              <Autocomplete
+                fullWidth
```

Comment:
📓 For some reason, Typescript complains about this value despite it existing in the docs and the definitions file.

Reply:
Strange, I wasn't seeing an issue with it.

Reply:
Yeah, I can't explain it unfortunately 🤷‍♂️. t was just showing up when I was adding the ts-ignores